test/src/test/java/hudson/model/JobTest.java

```diff
@@ -286,4 +291,35 @@ private static void tryConfigDotXml(JenkinsRule.WebClient wc, int status, String
         p.renameTo("different-name");
     }
 
+    @Issue("JENKINS-30502")
+    @Test public void testRenameTrimsSpace() throws Exception {
+        tryRename("myJob", " foo", "foo");
```

Comment:
This can degrade test time, and in some kind of logic this test relates to one piece of logic